websocket/src/test/java/org/nanohttpd/junit/protocols/websockets/WebSocketFrameTest.java

```diff
@@ -0,0 +1,188 @@
+            (byte) -1,
+            (byte) 0,
+            (byte) 0,
+            (byte) 0,
+            (byte) 1,
+            (byte) 47,
+            (byte) 120,
+            (byte) -16,
+            (byte) 16
+        };
+
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+    }
+
+    @Test
+    public void testReadWebSocketException4ByteLength() throws Exception {
+
+        // Arrange
+        final byte[] byteArray = {
+            (byte) -128,
+            (byte) -1,
+            (byte) -128,
+            (byte) 0,
+            (byte) 0,
+            (byte) 0,
+            (byte) 47,
+            (byte) 120,
+            (byte) -16,
+            (byte) 16
+        };
+
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+    }
+
+    @Test
+    public void testReadWebSocketException2ByteLength() throws Exception {
+
+        // Arrange
+        final byte[] byteArray = {
+            (byte) -59,
+            (byte) -128,
+            (byte) 126,
+            (byte) 0,
+            (byte) 0,
+            (byte) 32,
+            (byte) -1,
+            (byte) -128,
+            (byte) 4,
+            (byte) -128
+        };
+
+        // Arrange
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray, 1, 9);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+
+    }
+
+    @Test
+    public void testReadPayloadInfoReservedBitsNotZero() throws Throwable {
+
+        // Arrange
+        final byte[] byteArray = {
+            (byte) 24,
+            (byte) 127,
+            (byte) 0,
+            (byte) 0,
+            (byte) 0,
+            (byte) 0,
+            (byte) 0,
+            (byte) -128,
+            (byte) 0,
+            (byte) 1
+        };
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+        final ArrayList<WebSocketFrame> fragments = new ArrayList<WebSocketFrame>();
+        final OpCode opCode = OpCode.Close;
+        final WebSocketFrame objectUnderTest = new WebSocketFrame(opCode, fragments);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        objectUnderTest.read(in);
+
+    }
+
+    @Test
+    public void setTextPayload() throws WebSocketException, CharacterCodingException {
+
+        // Arrange
+        final OpCode opCode = OpCode.Text;
+        final ArrayList<WebSocketFrame> fragments = new ArrayList<WebSocketFrame>();
+        final String payload = "\u0001\u0000";
+
+        final WebSocketFrame objectUnderTest = new WebSocketFrame(opCode, fragments);
+
+        // Act
+        objectUnderTest.setTextPayload(payload);
+
+        // Assert side effects
+        Assert.assertTrue(objectUnderTest.getTextPayload().equals(payload));
```

Comment:
I want to double check i'm following the logic of the assertion here: 

We set the objectUnderTest using the webSocketFrame constructor 
We set the object to contain a payload string 
We retrieve this payload string
We check to see that the payload we have retrieved is the same as the payload we passed in 

I would imagine that we could change this assertion to `Assert.assertEquals(objectUnderTest.getTextPayload(), payload);`

However I do wonder, as you mentioned in the PR message, whether this is a worthwhile assertion. Perhaps this is made more valuable by finding a case in which we can `assertFalse` for the same `.getTxtPayload()` method?